### PR TITLE
Fix hover, autostyle button and dropdown

### PR DESCRIPTION
**Pull request by donflopez**

### CODE REVIEW

**Comment by xavijam** [comment]
I miss some tests

**Comment by donflopez** [comment]
Tests added

File: spec/widgets/dropdown/content-view.spec.js [hunk 1]
```diff
@@ -22,6 +22,17 @@ describe('widgets/dropdown/widget-dropdown-view', function () {
...
 
+  it('should not be the options button', function () {
+    $('.js-button').click();
+    expect($('.js-container').find('.js-editWidget').length).toBe(0);
+  });
+
+  it('should be the options button', function () {
+    this.model.set('show_options', true);
+    $('.js-button').click();
+    expect($('.js-container').find('.js-editWidget').length).toBe(1);
```
Comment (xavijam) [inline spec/widgets/dropdown/content-view.spec.js]: Edit and remove right?

File: spec/widgets/dropdown/content-view.spec.js [hunk 2]
```diff
@@ -22,6 +22,17 @@ describe('widgets/dropdown/widget-dropdown-view', function () {
     expect($('.js-container').find('.CDB-Dropdown').length).toBe(1);
   });
 
+  it('should not be the options button', function () {
+    $('.js-button').click();
+    expect($('.js-container').find('.js-editWidget').length).toBe(0);
```
Comment (xavijam) [inline spec/widgets/dropdown/content-view.spec.js]: Same here?

File: src/widgets/widget-model.js [hunk 3]
```diff
@@ -58,7 +58,7 @@ module.exports = cdb.core.Model.extend({
   isAutoStyleEnabled: function (autoStyle) {
     var styles = this.get('style');
 
-    if (!styles && (this.get('type') === 'category' || this.get('type') === 'histogram')) return true;
+    if ((!styles || !styles.auto_style) && (this.get('type') === 'category' || this.get('type') === 'histogram')) return true;
```
Comment (xavijam) [inline src/widgets/widget-model.js]: I'd add a test for this.

File: spec/widgets/dropdown/content-view.spec.js (hunk 1 above)
Comment (donflopez) [inline spec/widgets/dropdown/content-view.spec.js]: It is not the same, It checks that by default, the options buttons don't appear, and the other just the opposite.

Comment (xavijam) [inline spec/widgets/dropdown/content-view.spec.js]: Sorry, I don't get it. If options is true, both buttons should appear. If not, they should not appear.

Comment (donflopez) [inline spec/widgets/dropdown/content-view.spec.js]: Yeah! It's just as you've said! :)

Comment (xavijam) [inline spec/widgets/dropdown/content-view.spec.js]: So, why don't we check that both buttons are present?

Comment (donflopez) [inline spec/widgets/dropdown/content-view.spec.js]: Both are on the same conditional I thought it was redundant to check both of them.

Comment (xavijam) [inline spec/widgets/dropdown/content-view.spec.js]: Why not test there are 4 options instead of 2, not sure.

Comment (donflopez) [inline spec/widgets/dropdown/content-view.spec.js]: Yes, in that case... I'll check four options, two and if they are the corrects.

**Comment by donflopez** [comment]
Added all of them

**Pull request merged**